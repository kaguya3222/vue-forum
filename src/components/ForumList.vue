<template>
  <div>
    <forum-list-item
      v-for="forum in forums"
      :key="forum['.key']"
      :forum="forum"
    />
  </div>
</template>

<script>
import ForumListItem from "./ForumListItem";
export default {
  components: {
    "forum-list-item": ForumListItem
  },
  props: {
    forums: {
      required: true,
      type: Array
    }
  },
  name: "ForumList.vue"
};
</script>

<style scoped></style>
