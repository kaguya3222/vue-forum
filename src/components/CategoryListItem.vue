<template>
  <div class="forum-list">
    <h2 class="list-title">
      <router-link :to="{ name: 'Category', params: { id: category['.key'] } }">
        {{ category.name }}
      </router-link>
    </h2>
    <forum-list :forums="categoryForums" />
  </div>
</template>

<script>
import ForumList from "./ForumList";
import { mapGetters } from "vuex";

export default {
  name: "CategoryListItem.vue",
  components: { "forum-list": ForumList },
  props: {
    category: {
      required: true,
      type: Object
    }
  },
  computed: {
    ...mapGetters({
      forums: "forums/items"
    }),
    categoryForums() {
      return Object.values(this.forums).filter(
        forum => forum.categoryId === this.category[".key"]
      );
    }
  }
};
</script>

<style scoped></style>
