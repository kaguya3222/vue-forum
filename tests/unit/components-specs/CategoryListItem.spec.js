import { shallowMount } from "@vue/test-utils";
import CategoryListItem from "../../../src/components/CategoryListItem";
import ForumList from "../../../src/components/ForumList";
import mockedSourceData from "../mocks/mockedSourceData";

describe("CategoryListItem", () => {
  const wrapper = shallowMount(CategoryListItem, {
    propsData: {
      category: mockedSourceData.categories[0]
    },
    stubs: {
      "router-link": true
    }
  });
  test("Renders category name as a link", () => {
    const routerLinkStub = wrapper.find("router-link-stub");
    expect(routerLinkStub).toMatchSnapshot();
  });
  test("Accepts ForumList as a child component", () => {
    const forumList = wrapper.find(ForumList);
    expect(forumList.name()).toBe("ForumList.vue");
  });
  test("Creates forums array of certain category", () => {
    const forum = mockedSourceData.forums[0];
    const hasForum = wrapper.vm.categoryForums.some(
      categoryForum => categoryForum[".key"] === forum[".key"]
    );
    expect(hasForum).toBe(true);
  });
});
