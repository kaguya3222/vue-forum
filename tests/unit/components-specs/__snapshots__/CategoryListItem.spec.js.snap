// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryListItem Renders category name as a link 1`] = `
<router-link-stub to="[object Object]">
  Feedback &amp; Information
</router-link-stub>
`;
